source/code/providers/support/runasprovider.cpp

```diff
@@ -212,7 +219,11 @@ namespace SCXCore
 
         try
         {
-            SCXFilePath scriptfile = SCXFile::CreateTempFile(script);
+        	bool tmpDirExists = SCXCoreLib::SCXDirectory::Exists(m_defaultTmpDir);
+        	std::wstring tmpDir = tmpDirExists ? m_defaultTmpDir : L"/tmp/";
+        	if(!tmpDirExists)
+        		SCX_LOGWARNING(m_log, L"Default tmp Directory does not exist. Falling back to /tmp");
```

Comment:
A few minor nits here.

We have a coding standard that says you can't use a single-line if without {}. So change this to:

if(!tmpDirExists) {
          SCX_LOGWARNING(m_log, L"Default tmp Directory does not exist. Falling back to /tmp");
}

Next, if the system is set up wrong, we'll log this many times for each script run, filling the log file. This is undesirable. So, for this sort of thing, we prefer a one-time log message per agent start. We have a framework for that. See PAL routine SCXCoreLib::LogSuppressor, for am example see (from the PAL): source/code/scxsystemlib/disk/diskdepend.cpp. Please do a similar change here.